### ISSUE TITLE
Prevent KSP from running if TweakScale is installed on the wrong place!

### ISSUE THREAD

**Issue by kailGH** [net-lisias-ksp/TweakScale#77]
2.4.3.5 fails this check with its own fresh installation.

**Comment by Lisias** [comment]
Interesting…

Publish a screenshot of the message so I can check how it's calculating the file paths. Are you using Windows, Mac or Linux?

On Windows, you will need to do a Alt+PrintScreen and paste into Paintbrush to save it into a file.

**Comment by Lisias** [comment]
Hummm… wait…. I messed up the filepaths, this will not work on Windows… DAMNIT

**Comment by Lisias** [comment]
It was a bug on the KSPe Light.

 @kailGH , could you please download this file and give it a test drive? Replace the KSPe.Light.TwealScale.dll on your installment with this one.

[KSPe.Light.TweakScale.dll.zip](https://github.com/net-lisias-ksp/TweakScale/files/3718901/KSPe.Light.TweakScale.dll.zip)

**Comment by Lisias** [comment]
Well… The thing is still working on Mac. Need to confirm if it works on Windows now.

![Screen Shot 2019-10-11 at 15 32 02](https://user-images.githubusercontent.com/64334/66675827-8714f000-ec3c-11e9-8d27-210e7c7a2cf9.png)

**Comment by kailGH** [comment]
No sorry, that didn't seem to work.

![ksp](https://user-images.githubusercontent.com/6550808/66679725-d112cb80-ec66-11e9-88cb-abfc3ddf9577.png)

**Comment by Lisias** [comment]
Got it.  That confirms I'm an idiot. I completely forgot about the Windows' counter-slashes. :P

I'll work on this in the next hours, as soon the working hours end!

**Comment by kailGH** [comment]
Sorry about closing it (accident). Glad you have it sorted (Reopened).

**Comment by Lisias** [comment]
@kailGH , this is a link for the Release Candidate for 2.4.3.6 .

Can you confirm this is working now? I will publish the thing as long as I have confirmation from you!

(thanks for your report! this would cause some damage….)

**Comment by kailGH** [comment]
That seems to work now. Thanks.

**Comment by Lisias** [comment]
Great! :)

**Comment by kailGH** [comment]
Or not.. eek. Something may have gone wrong with the MM patching.. or it could be my previous attempts leaving some stuff behind. Checking..

![ksp2](https://user-images.githubusercontent.com/6550808/66684391-cd387680-ec71-11e9-9824-3ed9c005a93b.png)

**Comment by Lisias** [comment]
Now is a proper TweakScale FATAL Houston Message! :)

It's good news, things are working as I intended - TweakScale detected a lot of potential destructive problems on your installment (they were already there, by the way - were you using an older TweakScale release?) and is whinnying about.

Are you using SpannerMonkey's Add'Ons? If yes, please publish the KSP.log and ModuleManager.ConfigCache on Issue #62 .

Otherwise, publish them on [TweakScale's Thread](https://forum.kerbalspaceprogram.com/index.php?/topic/179030-*) so I can tackle down the problems there.

**Comment by kailGH** [comment]
I was using the latest & had 9 x sanity check warning message for a long time at start.

SpannerMonkey's addon's? Possibly, I have lots of addons (ahem). But, I don't recognise the name. 

Can I just post the KSP.LOG here?

**Comment by kailGH** [comment]
Most of it seems to be the "duplicated properties" (#34) issue.

**Comment by kailGH** [comment]
KSP.LOG: [KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/3719636/KSP.log)

**Comment by kailGH** [comment]
"unholy characters on the name" LOL

**Comment by kailGH** [comment]
The original 9 sanity checks are still there, now known as #13. The rest are mostly #34's (will check to see what that is later) & a couple of dozen unholy's (including 3 on Squad parts).

**Comment by Lisias** [comment]
Got it. You didn't deleted the previous version! Delete the TweakScale folder and install it again.

```
[LOG 21:57:36.096] Applying update TweakScale/patches/NFT_TweakScale/@PART[adapter-125-0625-2] to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
[LOG 21:57:49.462] Applying update TweakScale/patches/NF/NFC_TweakScale/@PART[adapter-125-0625-2] to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
[LOG 21:58:25.663] Applying update WarpPlugin/Patches/kspstockthermal/@PART[*adapter*]:FOR[WarpPlugin] to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
[LOG 21:58:49.606] Applying update InterstellarFuelSwitch/PatchManager/ActiveMMPatches/IntegratedDecoupler/@PART[*]:FINAL to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
[LOG 21:58:50.490] Applying update Kopernicus/Config/SolarPanels/@PART:FINAL to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
[LOG 21:58:52.203] Applying update PartInfo/PartInfo/@PART[*]:Final to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
[LOG 21:58:52.762] Applying update WarpPlugin/Patches/OreTanksFix/@PART[*]:FINAL to NearFutureConstruction/Parts/Adapters/adapter-125/adapter-125-0625-2.cfg/PART[adapter-125-0625-2]
```

**Comment by Lisias** [comment]
Dude, my apologies. **I WAS THE ONE BORKING ON THIS**.

I failed to merge correctly from the development branch, and the old, NFT_TweakScale.cfg file leaked into the distribution.

I didn't detect this because my testing process is way more robust on the development branch (for obvious reasons), and the distribution branch goes only through quick smoke tests. I failed foresee this situation.

I will issue a 2.4.3.7 release soon with this mess corrected.
